vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/impl/JcrPackageImpl.java

```diff
@@ -441,7 +447,36 @@ private void extract(ImportOptions options, boolean createSnapshot, boolean repl
                     }
                 }
                 if (p.isValid()) {
-                    subPacks.add(p);
+                    PackageId pId = p.getPackage().getId();
+                    String pName = pId.getName();
+                    Version pVersion = pId.getVersion();
+
+                    // get the list of packages available in the same group
+                    JcrPackageManager pkgMgr = new JcrPackageManagerImpl(s);                    
+                    List<JcrPackage> listPackages = pkgMgr.listPackages(pId.getGroup(), true);
+
+                    // keep some status variable if a more recent is found in the next loop
+                    boolean foundMoreRecent = false;
+                    JcrPackage foundPackage = null;
```

Comment:
- only remember the packageId of the found package.
- `foundPackage` is not very precise name. maybe: `newerPackageId`